packages/conform-react/helpers.ts

```diff
@@ -4,45 +4,55 @@ import {
 	VALIDATION_SKIPPED,
 } from '@conform-to/dom';
 import type { FieldConfig, Primitive } from './hooks.js';
-import type { CSSProperties, HTMLInputTypeAttribute } from 'react';
-
-interface FormControlProps {
-	id?: string;
-	name: string;
-	form?: string;
-	required?: boolean;
-	autoFocus?: boolean;
-	tabIndex?: number;
-	style?: CSSProperties;
-	'aria-describedby'?: string;
-	'aria-invalid'?: boolean;
-	'aria-hidden'?: boolean;
-}
-
-interface InputProps<Schema> extends FormControlProps {
-	type?: HTMLInputTypeAttribute;
-	minLength?: number;
-	maxLength?: number;
-	min?: Schema extends number ? number : string | number;
-	max?: Schema extends number ? number : string | number;
-	step?: Schema extends number ? number : string | number;
```

Comment:
All conform helpers are not only designed for native input element but also trying to be compatible with custom input. Although `min` / `max` support both number (e.g. `type=number`) and string (e.g. `type=datetime-local`), I have seen a component library with a number input expecting number only. This was the reason why I added those condition instead of relying on the types of the native input element.